pkg/cluster/workloadidentityresources_test.go

```diff
@@ -764,13 +775,234 @@ func TestGetPlatformWorkloadIdentityFederatedCredName(t *testing.T) {
+			},
+		},
+		{
+			name: "all identities match existing role assignments - noop",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					ClusterProfile: api.ClusterProfile{
+						ResourceGroupID: resourceGroupID,
+					},
+					PlatformWorkloadIdentityProfile: &api.PlatformWorkloadIdentityProfile{
+						PlatformWorkloadIdentities: map[string]api.PlatformWorkloadIdentity{
+							"1": {
+								ObjectID: objectId1,
+							},
+							"2": {
+								ObjectID: objectId2,
+							},
+						},
+					},
+				},
+			},
+			roles: map[string]api.PlatformWorkloadIdentityRole{
+				"1": {
+					OperatorName:     "1",
+					RoleDefinitionID: roleDefinitionId1,
+				},
+				"2": {
+					OperatorName:     "2",
+					RoleDefinitionID: roleDefinitionId2,
+				},
+			},
+			existingRoleAssignments: []mgmtauthorization.RoleAssignment{
+				{
+					Name: &roleName1,
+					RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+						Scope:            &resourceGroupID,
+						RoleDefinitionID: pointerutils.ToPtr(fmt.Sprintf("/subscriptions/%s%s", subscriptionId, roleDefinitionId1)),
+						PrincipalID:      &objectId1,
+					},
+				},
+				{
+					Name: &roleName2,
+					RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+						Scope:            &resourceGroupID,
+						RoleDefinitionID: pointerutils.ToPtr(fmt.Sprintf("/subscriptions/%s%s", subscriptionId, roleDefinitionId2)),
+						PrincipalID:      &objectId2,
+					},
+				},
+			},
+		},
+		{
+			name: "new required identity added - creates new roleassignment",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					ClusterProfile: api.ClusterProfile{
+						ResourceGroupID: resourceGroupID,
+					},
+					PlatformWorkloadIdentityProfile: &api.PlatformWorkloadIdentityProfile{
+						PlatformWorkloadIdentities: map[string]api.PlatformWorkloadIdentity{
+							"1": {
+								ObjectID: objectId1,
+							},
+							"2": {
+								ObjectID: objectId2,
+							},
+						},
+					},
+				},
+			},
+			roles: map[string]api.PlatformWorkloadIdentityRole{
+				"1": {
+					OperatorName:     "1",
+					RoleDefinitionID: roleDefinitionId1,
+				},
+				"2": {
+					OperatorName:     "2",
+					RoleDefinitionID: roleDefinitionId2,
+				},
+			},
+			existingRoleAssignments: []mgmtauthorization.RoleAssignment{
+				{
+					Name: &roleName1,
+					RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+						Scope:            &resourceGroupID,
+						RoleDefinitionID: pointerutils.ToPtr(fmt.Sprintf("/subscriptions/%s%s", subscriptionId, roleDefinitionId1)),
+						PrincipalID:      &objectId1,
+					},
+				},
+			},
+			wantAdded: []*arm.Resource{
+				workloadIdentityResourceGroupRBAC(stringutils.LastTokenByte(roleDefinitionId2, '/'), objectId2),
+			},
+		},
+		{
+			name: "identity object id replaced - deletes old roleassignment and creates new",
```

Comment:
The assumption in the implementation that there will always be 1:1 existence of objectID to role is always be going to hold true?
As per my reading of the code, it's going to be the case, as we delete the unwanted ones before and only then perform the addition.

Reply:
we've discussed this internally. If this assumption is no longer the case we're going to need more substantial changes (including a change to the PlatformWorkloadIdentityRoleSets struct definitions) anyways, so we could handle this then.